consensus/src/ordered_broadcast/mod.rs

```diff
@@ -459,18 +459,16 @@ mod tests {
                 }
                 context.sleep(Duration::from_millis(1000)).await;
                 *shutdowns.lock().unwrap() += 1;
-
-                context
             };
 
-            let context = if let Some(prev_ctx) = prev_ctx {
-                deterministic::Runner::from(prev_ctx)
+            let (_, checkpoint) = if let Some(prev_checkpoint) = prev_checkpoint {
+                deterministic::Runner::from(prev_checkpoint)
             } else {
                 deterministic::Runner::timed(Duration::from_secs(45))
             }
-            .start(f);
+            .start_and_recover(f);
```

Comment:
I find this a much saner way to "restart" the runtime (rather than having to pass `context` out of the scope it was created for).